File: app/src/main/java/com/simplexray/an/PrefsContract.java
package com.simplexray.an;

import android.net.Uri;
import android.provider.BaseColumns;

public final class PrefsContract {
    public static final String AUTHORITY = "com.simplexray.an.prefsprovider";
    public static final Uri BASE_CONTENT_URI = Uri.parse("content://" + AUTHORITY);
    public static final String PATH_PREFS = "prefs";

    private PrefsContract() {
    }

    public static final class PrefsEntry implements BaseColumns {
        public static final Uri CONTENT_URI =
                BASE_CONTENT_URI.buildUpon().appendPath(PATH_PREFS).build();
        public static final String CONTENT_TYPE =
                "vnd.android.cursor.dir/vnd." + AUTHORITY + "." + PATH_PREFS;
        public static final String CONTENT_ITEM_TYPE =
                "vnd.android.cursor.item/vnd." + AUTHORITY + "." + PATH_PREFS;
        public static final String COLUMN_PREF_KEY = "pref_key";
        public static final String COLUMN_PREF_VALUE = "pref_value";
        public static final String COLUMN_PREF_TYPE = "pref_type"; // To store type like String, int, boolean
    }
}

File: app/src/main/java/com/simplexray/an/PrefsContract.kt
package com.simplexray.an

import android.net.Uri
import android.provider.BaseColumns

object PrefsContract {
    const val AUTHORITY: String = "com.simplexray.an.prefsprovider"
    val BASE_CONTENT_URI: Uri = Uri.parse("content://$AUTHORITY")
    const val PATH_PREFS: String = "prefs"

    object PrefsEntry : BaseColumns {
        @JvmField
        val CONTENT_URI: Uri = BASE_CONTENT_URI.buildUpon().appendPath(PATH_PREFS).build()
        const val CONTENT_TYPE: String =
            "vnd.android.cursor.dir/vnd.$AUTHORITY.$PATH_PREFS"
        const val CONTENT_ITEM_TYPE: String =
            "vnd.android.cursor.item/vnd.$AUTHORITY.$PATH_PREFS"
        const val COLUMN_PREF_KEY: String = "pref_key"
        const val COLUMN_PREF_VALUE: String = "pref_value"
        const val COLUMN_PREF_TYPE: String = "pref_type"
    }
}
